CorsixTH/Lua/entities/humanoids/staff.lua

```diff
@@ -385,14 +385,20 @@ function Staff:checkIfNeedRest()
     if self.waiting_for_staffroom then
       self:changeAttribute("happiness", -0.001)
     end
+
     local room = self:getRoom()
-    if (self.staffroom_needed and ((room and not room:getPatient()) or not room)) or
-        (room and self.going_to_staffroom) then
-      if self:getCurrentAction().name ~= "walk" and self:getCurrentAction().name ~= "queue" then
-        self.staffroom_needed = nil
-        self:goToStaffRoom()
-      end
+    local may_leave = false
```

Comment:
Doesn't match the old condition and I think this is wrong now. `staffroom_needed` use to be restricted by the other conditions (no patient if in a room) `going_to_staffroom` was not.